### PR TITLE
[desktop] Add application name

### PR DESCRIPTION
**Pull request by endim8**
Also improve desktop entry & OSM auth dialog a bit

Screenshot from 2023-03-22 14-51-08

Screenshot from 2023-03-22 19-50-50

Screenshot from 2023-03-22 19-50-22

### CODE REVIEW

File: qt/CMakeLists.txt [hunk 1]
```diff
@@ -70,12 +70,12 @@ target_link_libraries(${PROJECT_NAME}
 
 if (BUILD_DESIGNER)
   target_link_libraries(${PROJECT_NAME} generator)
-  set(BUNDLE_NAME "OMaps.Designer")
-  set(BUNDLE_DISPLAY_NAME "OMaps Designer")
+  set(BUNDLE_NAME "Organic Maps.Designer")
```
Comment (biodranik) [inline qt/CMakeLists.txt:73]: Let's leave the same short binary name to avoid any changes in other repos.

File: qt/mainwindow.cpp [hunk 2]
```diff
@@ -634,7 +634,7 @@ void MainWindow::OnUploadEditsMenuItem()
   {
     auto & editor = osm::Editor::Instance();
     if (editor.HaveMapEditsOrNotesToUpload())
-      editor.UploadChanges(key, secret, {{"created_by", "OMaps " OMIM_OS_NAME}});
+      editor.UploadChanges(key, secret, {{"created_by", "Organic Maps " OMIM_OS_NAME}});
```
Comment (biodranik) [inline qt/mainwindow.cpp:637]: That change may complicate debugging for osmers.

File: qt/res/OrganicMaps.desktop [hunk 3]
```diff
@@ -2,12 +2,12 @@
...
 Name=Organic Maps
 Version=1.0
-GenericName=Detailed Offline Maps of the World
-GenericName[ru]=Подробная оффлайновая карта мира
+Comment=Detailed Offline Maps of the World
+Comment[ru]=Подробная оффлайновая карта мира
 Icon=organicmaps
-TryExec=OMaps
-Exec=OMaps
+TryExec=Organic\sMaps
```
Comment (biodranik) [inline qt/res/OrganicMaps.desktop:8]: Ditto

**Review by biodranik — COMMENTED** [review]
Can we change strings in the app without changing executable name?

File: qt/mainwindow.cpp (hunk 2 above)
Comment (endim8) [reply qt/mainwindow.cpp:637]: Mobile uses the full string "Organic Maps" eg. https://www.openstreetmap.org/changeset/133341165) so this will actually unify desktop with mobile :)

**Comment by endim8** [comment]
> Can we change strings in the app without changing executable name?

done 👍

File: qt/mainwindow.cpp [hunk 4]
```diff
@@ -131,7 +131,7 @@ MainWindow::MainWindow(Framework & framework, bool apiOpenGLES3,
   CreateNavigationBar();
   CreateSearchBarAndPanel();
 
-  QString caption = qAppName();
```
Comment (biodranik) [inline qt/mainwindow.cpp:134]: What is returned here?

File: qt/osm_auth_dialog.cpp [hunk 5]
```diff
@@ -62,15 +61,14 @@ OsmAuthDialog::OsmAuthDialog(QWidget * parent)
   vLayout->addLayout(buttonsLayout);
 
   setLayout(vLayout);
-  setWindowTitle(isLoginDialog ? tr(kLoginDialogTitle) : tr(kLogoutDialogTitle));
+  setWindowTitle(tr(kLoginDialogTitle));
 }
```
Comment (biodranik) [inline qt/osm_auth_dialog.cpp:65]: Why is this change necessary?

File: qt/res/OrganicMaps.desktop [hunk 6]
```diff
@@ -2,12 +2,12 @@
...
+Comment[ru]=Подробная оффлайновая карта мира
 Icon=organicmaps
 TryExec=OMaps
 Exec=OMaps
 Terminal=false
 StartupNotify=false
-Categories=Qt;Education;Science;Geography;Geoscience
-Keywords=Maps;Offline Maps;Minsk;London;Briefcase;Case;OMaps;OrganicMaps;OSM;OpenStreetMap;Map
+Categories=Maps;Education;Science;Geography;Geoscience;Qt
+Keywords=Map;Maps;Offline Maps;OMaps;OrganicMaps;OSM;OpenStreetMap
```
Comment (biodranik) [inline qt/res/OrganicMaps.desktop:13]:
```suggestion
Keywords=Map;Maps;Offline Maps;OMaps;OrganicMaps;Organic Maps;OSM;OpenStreetMap
```

File: qt/osm_auth_dialog.cpp (hunk 5 above)
Comment (endim8) [reply qt/osm_auth_dialog.cpp:65]: Before, the header would be "OpenStreetMap Login Dialog" or "Logout Dialog" depending on if the user is logged in. 
The dialog doesn't actually change though, so it makes more sense to always show it as "OpenStreetMap Login"

File: qt/res/OrganicMaps.desktop (hunk 6 above)
Comment (endim8) [reply qt/res/OrganicMaps.desktop]: fixed

File: qt/mainwindow.cpp (hunk 4 above)
Comment (biodranik) [reply qt/mainwindow.cpp:134]: What's in [applicationName](https://doc.qt.io/qt-5/qcoreapplication.html#applicationName-prop) ? For some reason, I don't see any docs about qAppName.

It would be great to take it somehow from CMake, to distinguish Designer build vs normal build.

**Comment by endim8** [comment]
ah! We never actually assigned an application name, so it fell back to the executable name. 
It also looks like qAppName() is just a macro for QApplication::ApplicationName().

now it will display "Organic Maps" by default, and "Organic Maps Designer" for designer

File: qt/main.cpp [hunk 7]
```diff
@@ -142,8 +142,13 @@ int main(int argc, char * argv[])
...
   // Pretty icons on HDPI displays.
   QApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);
-
   platform.SetupMeasurementSystem();
+  
+
+  QCoreApplication::setApplicationName("Organic Maps");
+#ifdef BUILD_DESIGNER
+  QCoreApplication::setApplicationName("Organic Maps Designer");
+#endif
```
Comment (biodranik) [inline qt/main.cpp:151]:
```suggestion
  
#ifdef BUILD_DESIGNER
  QCoreApplication::setApplicationName("Organic Maps Designer");
#else
  QCoreApplication::setApplicationName("Organic Maps");
#endif
```

Comment (endim8) [reply qt/main.cpp]: fixed, ty :)

**Comment by endim8** [comment]
huh... i just tried building after that change and it doesn't work any more :\ hang on

**Review by rtsisyk — APPROVED** [review]
LGTM

File: qt/res/OrganicMaps.desktop [hunk 8]
```diff
@@ -1,13 +1,11 @@
 [Desktop Entry]
 Type=Application
-Name=Organic Maps
 Version=1.0
-GenericName=Detailed Offline Maps of the World
-GenericName[ru]=Подробная оффлайновая карта мира
+Name=Organic Maps
+Comment=Detailed Offline Maps of the World
+Comment[ru]=Подробная оффлайновая карта мира
```
[the remark this answers is not in the document]
Comment (endim8) [reply qt/res/OrganicMaps.desktop:6]: according to the freedesktop spec, `Comment` is a tooltip, so this short one is good :)
https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html

**Comment by endim8** [comment]
@biodranik ptal :)
